lib/matplotlib/transforms.py

```diff
@@ -151,22 +151,6 @@ def __copy__(self):
                 other.set_children(val)  # val == getattr(other, key)
         return other
 
-    def __deepcopy__(self, memo):
-        # We could deepcopy the entire transform tree, but nothing except
-        # `self` is accessible publicly, so we may as well just freeze `self`.
-        other = self.frozen()
-        if other is not self:
-            return other
-        # Some classes implement frozen() as returning self, which is not
-        # acceptable for deepcopying, so we need to handle them separately.
-        other = copy.deepcopy(super(), memo)
-        # If `c = a + b; a1 = copy(a)`, then modifications to `a1` do not
-        # propagate back to `c`, i.e. we need to clear the parents of `a1`.
-        other._parents = {}
```

Comment:
I think this is the relevant problem: you don't want changes to the copy to invalidate dependents of the original instance.

Reply:
I _think_ your previous tests captured that case?
https://github.com/matplotlib/matplotlib/blob/b528d5ca7ef96f749e446a99bf7e6024814bbe64/lib/matplotlib/tests/test_transforms.py#L709
I left all of the previous tests in there. The override of the `__copy__()` method is still required to keep the parent/child references intact.

Reply:
Oh, I see.  This is because the default deepcopy is also going to deepcopy `_parents`, creating new dependents (that don't actually exist anywhere except as `_parents` entries, so they should in fact even get gc'ed out at some point?); in any case this means that we indeed don't need to manually clear out `_parents` like I previously did.

Reply:
:+1: I think my take on deepcopy is that it should be the same as if I were to pickle dump and then load, which we are already testing, so it should be feasible to achieve I think. Something like this:
```python
__deepcopy__(self):
    return pickle.loads(pickle.dumps(self))
```
But, we don't even need to do that in this case because the standard deepcopy (with this update) works on the figure object. Thats what I'm trying to get at in the added test and just show that we get new objects with the proper scales set from the base object.